### ISSUE TITLE
BUG: RCNN_proposal_target need to be skipped for eval() mode

### ISSUE THREAD

**Issue by lsqshr** [jwyang/faster-rcnn.pytorch#387]
In `fasterRCNN.forward`, the roi sampling contained in `RCNN_proposal_target` gives random results which does not make sense when the model is in eval mode. 

RCNN_proposal_target should only be executed only when `self.training` is True
